libraries/AP_OpticalFlow/AP_OpticalFlow_UPFLOW.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+   driver for UPixel UPFLOW optical flow sensor
+
+   UPFLOW serial packet description
+   byte0: header0 (0xFE)
+   byte1: header1 (0x0A)
+   byte2: x-motion low byte;
+   byte3: x-motion high byte;
+   byte4: y-motion low byte;
+   byte5: y-motion high byte;
+   byte6: dt low byte;
+   byte7: dt high byte;
+   byte8: reserved;
+   byte9: reserved;   
+   byte10: surface quality
+   byte11: hardware version
+   byte12:checksum
+   byte13:footer (0x55)
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_OpticalFlow_UPFLOW.h"
+#include <AP_Math/crc.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <utility>
```

Comment:
I think we can remove some of these includes right?  I think this probably happened because you used the CXOF driver as a base (good idea!) but let's remove the ones that we don't need for this new driver, thanks!